src/server/uds.rs

```diff
@@ -0,0 +1,233 @@
+use std::fmt;
+use std::io;
+use std::os::unix::net::{SocketAddr
+    as UnixSocketAddr, UnixListener as StdUnixListener};
+use std::time::{Duration, Instant};
+
+use futures::{Async, Future, Poll, Stream};
+use tokio_reactor::Handle;
+use tokio_uds::UnixListener;
+use tokio_timer::Delay;
+use std::path::Path;
+
+use self::addr_stream::AddrStream;
+
+/// A stream of connections from binding to an address.
+#[must_use = "streams do nothing unless polled"]
+pub struct UnixAddrIncoming {
```

Comment:
I'd change this to just `UnixIncoming`.

Reply:
`std` calls it a `UnixStream` (for `TcpStream`), so I wouldn't feel right about using the same name for something else. Maybe `PathIncoming`? That is, for consistency, where hyper's tcp implementation calls it "AddrSomething", we call this "PathSomething".

Reply:
But std doesn't define a `UnixIncoming` does it?

Reply:
std doesn't define `TcpIncoming` either :) - std has `std::os::unix::net::SocketAddr` and `std::net::SocketAddr` which is what the hyper's tcp `AddrIncoming` is supposed to wrap.